nathanielmillard/rancid-tomatillos
# Nm favoriting

#### What’s this PR do?
-Creates a favorite button component with two visual displays 
-Connects to the local micro service to allow favorite status of a movie to persist between page loads
-Displays the favorite button component on both the MovieMain and MovieShowPage sections when a user is logged in
-Creates a new route for a favorite movies section 
-Adds a sad path to insure users sign in before seeing this section if they manage to navigate there without being signed in

#### Where should the reviewer start?
- Git pull this branch.
- Ensure that you also have the rancid-tomatillos-microserver [located here](https://github.com/nathanielmillard/rancid-tomatillos-microserver).

#### How should this be manually tested?
- Once you are running the microserver, run 'npm start' for this repo
- Log in as 'lucy@turing.io' with the password 'password1'
- Now you should see favorite icons on both the MovieMain and MovieShowPages.
- Click on the icons to see their status toggle back and forth
- Once you have some faovrites navigate to he faovirtes page
- You should be able to see that in live time unfoavoriting a movie will rerender this section without that movie tile

#### Any background context you want to provide?
-I am currenlty working on a branch for a final sweep/comb over things where I am refactoring and simplifying code where I see fit, really double checking PropTypes, and insuring we are only passing what props we absolutly need.
-There may be some stray comments, console logs, and things I am cleaning up on this next sweep.

#### What are the relevant tickets?
-closed #44 
-closes #45 

#### Screenshots (if appropriate):

#### Questions:
